### DISCUSSION TITLE
Multi Language Support (including Rulebook Import)

### DISCUSSION THREAD

**Discussion by Joly0** [fmasa/wfrp-master#45] [Ideas]
I have thought about the multi language support regarding various localized rulebooks and came up with an idea that might partially help. My idea would be to include the deepl api into the process, so users could import an english rulebook and the deepl api could maybe be used to translate the imported file afterwards if needed. The idea is not limited to deepl, google translate could be used aswell or other services, though i have had good results with translation using deepl,in comparision to google translate.

The only problem here is, that one would need a deepl api key, which is generally free, though can only translate 5k characters in the free tier, regarding the other tiers, i have no idea, on how to do that regarding the pricing and usage. But in either case, the user would have to bring their own api key.

It would be easier, if the translation could be directly in the app, though i am not sure if this is legally possible if we would take an english rulebook and translate it to other languages and distribute that through the app.

**Comment by fmasa** [comment]
> It would be easier, if the translation could be directly in the app, though i am not sure if this is legally possible if we would take an english rulebook and translate it to other languages and distribute that through the app.

Keeping copyrighted texts inside the app is not possible.

---

Apart from that I don't think that using translators is the way to go as there are rules-specific terms (Fortune Points, Wounds, etc) that are context-dependent on rulebook as they may have different meaning outside the WFRP context.

I would love to improve upon the JSON import/export feature as I think that scraping the PDF is questionable from copyright point of view as well. This would let users bring their own content.
